timesketch/api/v1/resources/sigma.py

```diff
@@ -92,10 +120,255 @@ def get(self, rule_uuid):
         meta = {"current_user": current_user.username, "rules_count": len(sigma_rules)}
         return jsonify({"objects": [return_rule], "meta": meta})
 
-
+# TODO(jaegeral): deprecate this class
 class SigmaByTextResource(resources.ResourceMixin, Resource):
     """Resource to get a Sigma rule by text."""
 
+    @login_required
+    def post(self):
+        """Handles POST request to the resource.
+        Returns:
+            JSON sigma rule
+        """
+
+        form = request.json
+        if not form:
+            form = request.data
+
+        content = form.get("content")
+        if not content:
+            return abort(
+                HTTP_STATUS_CODE_BAD_REQUEST, "Missing values from the request."
+            )
+
+        try:
+            sigma_rule = ts_sigma_lib.get_sigma_rule_by_text(content)
+
+        except ValueError:
+            logger.error(
+                "Sigma Parsing error with the user provided rule", exc_info=True
+            )
+            abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                "Error unable to parse the provided Sigma rule",
+            )
+
+        except NotImplementedError as exception:
+            logger.error(
+                "Sigma Parsing error: Feature in the rule provided "
+                " is not implemented in this backend",
+                exc_info=True,
+            )
+            abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                "Error generating rule {0!s}".format(exception),
```

Comment:
In line with my comment above, I think the error message that gets sent back to the user could be a bit more verbose, don't you think? "Error generating rule: NotImplementedError" is a very obscure error message to present to a client.